### DISCUSSION TITLE
Welcome to IntexPureSpa Discussions!

### DISCUSSION THREAD

**Discussion by Yogui79** [Yogui79/IntexPureSpa#1] [General]
## 👋 Hello and Welcome
  This discution is to speak about Intex PurerSpa remote control over Arduino or ESP32 and LC12S
   - if you have some question for installation fell free to ask 
   - if you have some idea for this code your idea are welcome

**Comment by struys** [comment]
Hi there! My friend Misko and I have been working on basically writing [the same code](https://github.com/misko/lc12s_py/blob/main/lc12s.py) for this [model](https://www.target.com/p/intex-28439e-greywood-deluxe-4-person-inflatable-hot-tub-bubble-jet-spa-grey/-/A-79574772?ref=OpsEmail_Order_280&j=168777&sfmc_sub=379792746&l=20_HTML&u=93080388&mid=7284873&jb=760535).

How did you figure out the network ids for your models?

**Reply by Yogui79** [reply]
hello,

you never answhere my question on the youtube post ;-) (Paul Personnes)

your spa look like  [mine](https://www.intex.fr/shop/spa-gonflable/3841-28442ex-spa-gonflable-baltik-6-places-6941057418230.html) but with an other reference

I use a channel scanner to find the chanel with help from an NRF24L01+ and an arduino . becarfull chanel change somtime but if you are very close to the controler 2 or 3 chanel works (I work on it since about 9 month and activly with  @LEECHER1 thins 3 mounth) 

after that I try all 128 network ID until I recive some message from the pump controler or from the remote control.

in my code you will found in the function SetSettings  the network id  maybe you can try it

```
//Network ID
#ifdef _28442_28440_  
  Config[5]=0xFF;
  Config[6]=0XFF;
#endif

#ifdef _28458_28462_  
  Config[5]=0x0D;
  Config[6]=0X05;
#endif
```
   
maybe your friend @misko  can look inside my code 
if found time I will try with a Raspberry also

**Reply by Yogui79** [reply]
@struys 
could you please invite your friend to this discution. we kwnow jet we have an issue with the network id . he change for eatch spa. it will be ineresting for evrybody we share our knowledge about this point

**Reply by misko** [reply]
Hi! that sounds difficult! We did not open up the spa controller, we determined network id and channel from wireless. 
I think we can optimize this a bit more , but here is what we did :) 
1) LED on LC12S blinks when you are on same channel regardless of network ID 
2) To find channel, iterate through channels and wait for 1second on each channel, if LED does not blink, then this is not the right channel (LED blinks regardless of network ID , only depends on channel)
3) Once correct channel is found (~1minute) , then scan for network ID (~7hrs)

To automate the LED blinking, we unsoldered the LED and connected LED signal to a GPIO pin on the raspberry pi. We scan for network ID like this, https://github.com/misko/lc12s_py/blob/main/lc12s.py#L253 and we scan for channel by using this https://github.com/misko/lc12s_py/blob/main/lc12s.py#L226

We just managed to capture our first data the other day, we are storing captured packets in binary files and now a new part of the code can read binary files and manage signaling. https://github.com/misko/lc12s_py/blob/main/spa.py
A lot of this file is based on your code!!!! Without it we would be seriously in trouble :( 

Thank you! what are you thinking for network ID? can you scan for it? How does manufacturer send new remotes when remote is damaged, does it come preprogrammed with networkID?

**Reply by misko** [reply]
Network ID we know for @struys spa is 0xC11E and current channel is 0x34 , the command_char seems to be 0x34+0x80 (maybe we are off by one channel and match your 0x7F from the code?)

**Reply by Yogui79** [reply]
hello @misko

we make the same (only visual) to found the chanel and the network id just put it insde an other arduino code.

we detect somme chanel change over the time thats why I make a chanel check  . dependig on the spa model (without jet) I also detect the command char calculation or not.

I had some case were  command_char  was chanel from 0x7C up to 0x80 I havent found any reason for that, thats why I make a second test after the chanel check 

to find the chanel we have up to 3 min (0-0xFF) we have chanel in range 0x4y for the network id its depending,mine is 0xFFFF and I ned more or less 2 days to found it  

regarding the management from the networkid @LEECHER1  and myself are juste making some test to try to found a way du catch it.one is sure the network id is not given from the controler but from the pump we have 2 controler and we are able to switch without network id change

**Comment by Fheodoor01** [comment]
Hi! First of all, thanks for the very nice code! I got everything to work, but have no idea how to "import" the config file into Home Assistant? (Just installed it, so no experience there ;-) ). Could you give me some pointers? (Maybe add to manual). Thanks a lot!

Ps: One suggestion, probably in newer iterations, you use Wifi.h library (for OTA), but it can conflict with the MQTT with integrated wifi handling. In fact I might be better to just use a plain MQTT handler after connecting with wifi (trough Wifi.h)

**Reply by Fheodoor01** [reply]
Found it! Apologies for bothering. Although saving the complete interface would be nice, but that's more a job of HA to fix.

**Reply by VictorHugoAguilar** [reply]
some reference

**Comment by VictorHugoAguilar** [comment]
some reference

**Reply by VictorHugoAguilar** [reply]
any
